### PR TITLE
Add `isSatisfiedByAny()` to `Range` and `RangesList`

### PR DESCRIPTION
**Pull request by sschuberth**
These convenience methods check whether a range was created in such a way that it matches any version.

### CODE REVIEW

**Comment by piotrooo** [comment]
Many thanks 🙏 
I'll review this when I'll be on PC.

File: src/main/java/org/semver4j/Range.java [hunk 1]
```diff
@@ -26,6 +26,10 @@ public Semver getRangeVersion() {
         return rangeVersion;
     }
 
+    public boolean isSatisfiedByAny() {
+        return rangeVersion.isEqualTo("0.0.0") && rangeOperator == RangeOperator.GTE;
```
Comment (piotrooo) [inline src/main/java/org/semver4j/Range.java:30]: Maybe we should extract this "magic" string `0.0.0` to some const. We use it in https://github.com/semver4j/semver4j/blob/87d98cabd6fde09efe01138622468b11e8c52566/src/main/java/org/semver4j/internal/range/processor/GreaterThanOrEqualZeroProcessor.java#L20

I propose in `Semver` class:

https://github.com/semver4j/semver4j/blob/87d98cabd6fde09efe01138622468b11e8c52566/src/main/java/org/semver4j/Semver.java#L17-L19

add const:

```
public static final Semver ZERO = new Semver("0.0.0");
```

This allows us to use it in `isSatisfiedByAny` method:

```
public boolean isSatisfiedByAny() {
    return rangeVersion.isEqualTo(ZERO) && rangeOperator == RangeOperator.GTE;
}
```

And in the `GreaterThanOrEqualZeroProcessor` like this:

```
return format(Locale.ROOT, "%s%s", GTE.asString(), ZERO);
```

File: src/main/java/org/semver4j/RangesList.java [hunk 2]
```diff
@@ -44,6 +44,11 @@ public RangesList add(List<Range> ranges) {
         return this;
     }
 
+    public boolean isSatisfiedByAny() {
+        return rangesList.stream().flatMap(List::stream)
+            .allMatch(range -> range.isSatisfiedByAny());
```
Comment (piotrooo) [inline src/main/java/org/semver4j/RangesList.java:49]: Use fluent method in new line and replace lambda with method reference:

```
public boolean isSatisfiedByAny() {
    return rangesList.stream()
        .flatMap(List::stream)
        .allMatch(Range::isSatisfiedByAny);
}
```

File: src/test/java/org/semver4j/RangeTest.java [hunk 3]
```diff
@@ -14,6 +14,15 @@
 import static org.semver4j.Range.RangeOperator.*;
 
 class RangeTest {
+    @Test
+    void shouldCheckSatisfiedByAny() {
+        //given
+        Range range = new Range("0.0.0", GTE);
```
Comment (piotrooo) [inline src/test/java/org/semver4j/RangeTest.java:20]: When we extract a `ZERO` const, could be used here.

**Review by piotrooo — CHANGES_REQUESTED** [review]

**Comment by piotrooo** [comment]
@sschuberth gently ping :smile:

**Comment by sschuberth** [comment]
> @sschuberth gently ping 😄

Sorry, I was busy at my day job.

**Review by piotrooo — DISMISSED** [review]

**Comment by sschuberth** [comment]
@piotrooo I had to adjust the initialization order of the static `ZERO` field.

**Comment by sschuberth** [comment]
> @piotrooo I had to adjust the initialization order of the static `ZERO` field.

Yet again 😒

**Comment by piotrooo** [comment]
@sschuberth thanks. I've merge it with errors but I need to fix it by my own.

**Pull request merged**